src/oop/special_class_method/medium/q12/MainTest.java

```diff
@@ -0,0 +1,144 @@
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"), String.format(incorrectModifierMessage, getAttributeMethodName1, className));
+        assertTrue(testClass.hasReturnType(getAttributeMethodName1, null, int.class), String.format(incorrectReturnType, getAttributeMethodName1, className));
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                "Your " + getAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        _assertEquals(value1, getMethodOutput,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the " + attributeName1 + " attribute.");
+    }
+
+    @Test
+    public void displaySizeIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName, null), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(testClass.hasModifier(methodName, null, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(testClass.hasReturnType(methodName, null, Void.TYPE), String.format(incorrectReturnType, methodName, className));
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctDisplaySizeMethod(int value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, int.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName, null), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(testClass.hasModifier(methodName, null, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(testClass.hasReturnType(methodName, null, Void.TYPE), String.format(incorrectReturnType, methodName, className));
+
+        if (value2 > 2000) {
+            testClass.callMethod(methodName, testInstance, new Clause[]{
```

Comment:
You can also define a custom error message here for if it prints the correct output.